prometheus/src/snapshot_metrics.rs

```diff
@@ -0,0 +1,65 @@
+use crate::utils::{write_metric, Metric, MetricFamily};
+use solana_runtime::snapshot_archive_info::{SnapshotArchiveInfo, SnapshotArchiveInfoGetter};
+use solana_runtime::snapshot_config::SnapshotConfig;
+use solana_runtime::snapshot_utils;
+use solana_sdk::clock::Slot;
+use std::io;
+
+pub fn write_snapshot_metrics<W: io::Write>(
+    snapshot_config: &Option<SnapshotConfig>,
+    out: &mut W,
+) -> io::Result<()> {
+    let snapshot_config = match snapshot_config.as_ref() {
+        Some(config) => config,
+        None => return Ok(()),
+    };
+
+    let full_snapshot_info = match get_full_snapshot_info(snapshot_config) {
+        Some(info) => info,
+        None => return Ok(()),
+    };
+    write_metric(
+        out,
+        &MetricFamily {
+            name: "solana_full_snapshot_info",
+            help: "The slot height of the most recent full snapshot",
+            type_: "counter",
+            metrics: vec![Metric::new(full_snapshot_info.slot)
+                .with_label("hash", full_snapshot_info.hash.to_string())],
+        },
+    )?;
+
+    // Incremental snapshots may be disabled, so we write full snapshot
+    // metric before and just return early if that is the case.
+    let incremental_snapshot_info =
+        match get_incremental_snapshot_info(snapshot_config, full_snapshot_info.slot) {
+            None => return Ok(()),
+            Some(info) => info,
+        };
+    write_metric(
+        out,
+        &MetricFamily {
+            name: "solana_incremental_snapshot_info",
+            help: "The slot height of the most recent incremental snapshot",
+            type_: "counter",
+            metrics: vec![Metric::new(incremental_snapshot_info.slot)
+                .with_label("hash", incremental_snapshot_info.hash.to_string())],
+        },
+    )
+}
+
+fn get_full_snapshot_info(snapshot_config: &SnapshotConfig) -> Option<SnapshotArchiveInfo> {
```

Comment:
Do we need the clone here, can’t we return a reference?

Reply:
Well, we do not need to but then we cannot have it as a separate function as we would return reference to variable scoped in this func. I originally did it that way, reverted it back to avoid cloning.